src/qibocal/cli/compare.py

```diff
@@ -118,34 +118,119 @@ def merge_plots(self, plots: list[list[go.Figure]]) -> list[go.Figure]:
         Scatter plots are plotted in the same figure.
         Heatmaps are vertically stacked.
         """
-        if plots[0][0].data[0].type == "scatter":
-            merged_plot_data = plots[0][0].data
-            for plot in plots[1:]:
-                merged_plot_data = merged_plot_data + plot[0].data
-            fig = [go.Figure(data=merged_plot_data, layout=plots[0][0].layout)]
-        elif plots[0][0].data[0].type == "heatmap":
-            fig = make_subplots(rows=2, cols=2)
-            for i, plot in enumerate(plots):
-                fig.append_trace(
-                    plot[0].data[0],
-                    row=i + 1,
-                    col=1,
-                )
-                fig.append_trace(
-                    plot[0].data[1],
-                    row=i + 1,
-                    col=2,
+        for i in range(len(plots[0])):
+            fig0 = plots[0][i]
+            fig1 = plots[1][i]
```

Comment:
Minor remark.
Can we zip `plots[0]` and `plots[1]`? It removes a variable and it's safer (even though we expect `plots[0]` and `plots[1]` to have the same length). We can either save the combined plot into a new list or we can reuse `fig0`.
```suggestion
        for fig0, fig1 in zip(plots[0], plots[1]):
```

Reply:
Implemented in https://github.com/qiboteam/qibocal/pull/937/commits/366b6327d9195a4f63ea6e493ef9e001a06a160b
I've also changed from `append_trace` to `add_trace` since I believe that for what we are doing the result should be the same.
If it looks good to you feel free to merge @GabrielePalazzo